core/src/main/java/org/arquillian/cube/impl/client/container/ContainerConfigurationController.java

```diff
@@ -64,12 +67,20 @@ public void remapContainer(@Observes BeforeSetup event, CubeRegistry cubeRegistr
                     int containerPort = getDefaultPortFromConfigurationInstance(newConfigurationInstance,
                             configurationClass, configurationClassPortField);
 
-                    PortBinding bindingForExposedPort = null;
-                    if ((bindingForExposedPort = binding.getBindingForExposedPort(containerPort)) != null) {
+                    PortBinding bindingForPort = null;
+                    if ((bindingForPort = binding.getBindingForExposedPort(containerPort)) != null) {
```

Comment:
Correct. configuredBinding() is what we could read out from configuration, e.g. docker-compose/cube format/arquillian.xml/pod.json etc. There is a case for when we need to reconfigure a DeployableContainer to use remapped ports, we need to do that Before the DeploymentContainer is setup, which is Before the Cube is started.